frame/support/src/traits/hooks.rs

```diff
@@ -417,16 +436,24 @@ mod tests {
 	#[cfg(feature = "try-runtime")]
 	#[test]
 	fn on_runtime_upgrade_tuple() {
+		use sp_io::TestExternalities;
+		use sp_std::sync::atomic::{AtomicUsize, Ordering};
+
 		struct Test1;
 		struct Test2;
 		struct Test3;
 
+		static TEST1_ASSERTIONS: AtomicUsize = AtomicUsize::new(0);
```

Comment:
not really looking deeply, but can't you use `thread_local`? in which case you can use `parameter_types`.